packages/dashboard/src/hooks/useAssetModel/useAssetModel.ts

```diff
@@ -0,0 +1,66 @@
+import { IoTSiteWiseClient } from '@aws-sdk/client-iotsitewise';
+import { useQueries, type QueryFunctionContext } from '@tanstack/react-query';
+import invariant from 'tiny-invariant';
+import { AssetModelCacheKeyFactory } from './assetModelCacheKeyFactory';
+import { DescribeAssetModelRequest } from './describeAssetModelRequest';
+import { createNonNullableList } from '~/helpers/lists/createNonNullableList';
+
+type SingleAssetRequest = {
+  assetModelId?: string;
+  assetModelIds?: never;
+};
+
+type MultiAssetRequest = {
+  assetModelIds?: string[];
+  assetModelId?: never;
+};
+
+export type UseAssetModelOptions = {
+  client: IoTSiteWiseClient;
+} & (SingleAssetRequest | MultiAssetRequest);
+
+/** Use the list of child assets for an asset with a given asset ID. */
+export function useAssetModel({ assetModelId, assetModelIds, client }: UseAssetModelOptions) {
```

Comment:
we use arrow function notation elsewhere in the code, so probably would be nice to change these to arrow functions

Reply:
I already broke this paradigm in the query editor... lol

IMO I prefer to use function declarations as the default and use arrows when necessary. Does it not save you a few seconds knowing something is a function with the word "function" at the beginning instead of having to look at the end of the line to know it's a function and not a different data type?

Reply:
I don't have an opinion on arrow vs function notation, but I wanted to follow the pattern used in the other iot sitewise client hooks